src/module.c

```diff
@@ -8455,8 +8453,7 @@ void moduleBlockedClientTimedOut(client *c, int from_module) {
 
     moduleFreeContext(&ctx);
 
-    if (!from_module)
-        updateStatsOnUnblock(c, bc->background_duration, 0, server.stat_total_error_replies != prev_error_replies);
+    updateStatsOnUnblock(c, bc->background_duration, 0, server.stat_total_error_replies != prev_error_replies);
```

Comment:
but `updateStatsOnUnblock()` is not thread-safe, it can't be called from main thread and module thread at the same time.

Reply:
mmm well then there's a flaw in the design
moduleBlockedClientTimedOut can be called either from the main thread or from a module thread (but on if the client is blocked on keys)... but both need to update stats

i can revert this change and simply turn off `bc->blocked_on_keys` in case of RM_UnblockClient

then `moduleHandleBlockedClients` will call `updateStatsOnUnblock` for that cleint